torch/_inductor/codegen/cpp_utils.py

```diff
@@ -76,6 +86,143 @@
 GemmBlocking = namedtuple("GemmBlocking", ["block_m", "block_n", "block_k"])
 
 
+def get_promote_dtype(args):
+    return (
+        functools.reduce(
+            torch.promote_types,  # type: ignore[arg-type]
+            [n.dtype for n in args if isinstance(n, CppCSEVariable)],
+        )
+        if all(n.dtype is not None for n in args if isinstance(n, CppCSEVariable))
+        else None  # not enough info to calculate the promote dtype
+    )
+
+
+def promote_vec_args(new_args):
```

Comment:
What's the diff from promoting normal `CppCSEVariable` args that could be either scalars or vectors?

Reply:
yes, I think we can do the `promote_args` for `scalars` and `vectors` unified and changed.